tracer/test/Datadog.Trace.Tests/Util/Http/QueryStringObfuscatorTests.cs

```diff
@@ -67,7 +67,7 @@ public void ObfuscateWithCustomPattern()
         public void ObfuscateTimeout()
         {
             var logger = new Mock<IDatadogLogger>();
-            var obfuscator = ObfuscatorFactory.GetObfuscator(100, @"\[(.*?)\]((?:.\s*)*?)\[\/\1\]", logger.Object);
+            var obfuscator = ObfuscatorFactory.GetObfuscator(100, @"\[(.*?)\]((?:.\s*)*?)\[\/\1\]", logger.Object, supportBacktracking:true);
```

Comment:
We should definitely test with both options.

Reply:
The regex we test here is an example crafted to cause a timeout, it's not a regex we're going to use. The regex throws an error with the non-backtracking, as this particular case isn't supported, so the test was throws the wrong exception type.

I think crafting a regex non-backing regex that would consistently cause a timeout would be hard. On reflection adding another options to provoke a timeout, so the test can work isn't a good idea, but I'm not really sure how to fix this test in that case. Perhaps it can be removed?

Reply:
would it be possible to test for `RegexMatchTimeoutException` when `supportBacktracking: true` and the other exception type (`NotSupportedException`?) when `supportBacktracking:false`?